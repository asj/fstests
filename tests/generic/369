#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright 2024 Google LLC
#
# FS QA Test No. 369
#
# Verify the ciphertext for encryption policies that use a hardware-wrapped
# inline encryption key, the IV_INO_LBLK_32 flag, and AES-256-XTS.
#
. ./common/preamble
_begin_fstest auto quick encrypt

. ./common/filter
. ./common/encrypt

# Hardware-wrapped keys require the inlinecrypt mount option.
_require_scratch_inlinecrypt
export MOUNT_OPTIONS="$MOUNT_OPTIONS -o inlinecrypt"

_verify_ciphertext_for_encryption_policy AES-256-XTS AES-256-CTS-CBC \
	v2 iv_ino_lblk_32 hw_wrapped_key

status=0
exit
