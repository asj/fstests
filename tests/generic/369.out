QA output created by 369

Verifying ciphertext with parameters:
	contents_encryption_mode: AES-256-XTS
	filenames_encryption_mode: AES-256-CTS-CBC
	options: v2 iv_ino_lblk_32 hw_wrapped_key
